From 739c89319601b303d523d14a1b6858d3bea8789c Mon Sep 17 00:00:00 2001
From: Matt Fleming <mfleming@suse.de>
Date: Tue, 31 Oct 2017 13:57:03 +0000
Subject: [PATCH] sched/fair: Enable SIS_AVG_CPU by default
Patch-mainline: Not yet, pending
References: bsc#1061593

This scheduler knob was introduced to limit scans of the LLC domain in
situations where the cost of scanning isn't worth it.

Turning it on results in a decent improvement to tbench's throughput
for all machines on the performance test grid.

Here's a typical result:

tbench4
                                   4.12.14                4.12.14
                                    sles15     sles15-sis-avg-cpu
Hmean     mb/sec-1        568.89 (   0.00%)      581.57 (   2.23%)
Hmean     mb/sec-2       1136.19 (   0.00%)     1126.80 (  -0.83%)
Hmean     mb/sec-4       2114.17 (   0.00%)     2155.82 (   1.97%)
Hmean     mb/sec-8       3868.19 (   0.00%)     3900.22 (   0.83%)
Hmean     mb/sec-16      7355.25 (   0.00%)     7378.81 (   0.32%)
Hmean     mb/sec-32     12610.86 (   0.00%)    12633.55 (   0.18%)
Hmean     mb/sec-64     16523.71 (   0.00%)    21391.61 (  29.46%)
Hmean     mb/sec-128    31329.89 (   0.00%)    31723.43 (   1.26%)
Hmean     mb/sec-256    31611.06 (   0.00%)    31977.63 (   1.16%)
Hmean     mb/sec-320    31773.01 (   0.00%)    31836.25 (   0.20%)

Signed-off-by: Matt Fleming <mfleming@suse.de>
---
 kernel/sched/features.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/features.h b/kernel/sched/features.h
index 319ed0e8a347..01125351736e 100644
--- a/kernel/sched/features.h
+++ b/kernel/sched/features.h
@@ -54,7 +54,7 @@ SCHED_FEAT(TTWU_QUEUE, true)
 /*
  * When doing wakeups, attempt to limit superfluous scans of the LLC domain.
  */
-SCHED_FEAT(SIS_AVG_CPU, false)
+SCHED_FEAT(SIS_AVG_CPU, true)
 SCHED_FEAT(SIS_PROP, true)
 
 /*
-- 
2.14.2

